webpack/assets/javascripts/react_app/components/AutoComplete/AutoCompleteActions.js

```diff
@@ -0,0 +1,97 @@
+import URI from 'urijs';
+import API from '../../API';
+import { STATUS } from '../../constants';
+import {
+  AUTO_COMPLETE_REQUEST,
+  AUTO_COMPLETE_SUCCESS,
+  AUTO_COMPLETE_FAILURE,
+  AUTO_COMPLETE_RESET,
+  TRIGGERS,
+} from './AutoCompleteConstants';
+
+export const getResults =
+(path, searchQuery, controller, trigger, queryCache) => (dispatch) => {
+  const loadNextResults = trigger === TRIGGERS.ITEM_SELECT ? '+' : '';
+  const url = `${path}${URI.encodeQuery(searchQuery)}${loadNextResults}`;
+  // to avoid an extra API call, find if this search query was already used.
+  const usedQueryResults = queryCache[controller] && queryCache[controller][searchQuery];
```

Comment:
There is a problem with the caching, I think.

Scenario:
1) open two tabs:
  a) hosts index page
  b) hosts create page
2) search a specific host name in the first tab
3) create a new host in the second one
4) in the first tab searching for the host that I just created by name is not shown...

Reply:
yeah, you absolutely right.. I will think of something..